framework/base/Module.php

```diff
@@ -571,32 +571,43 @@ public function createController($route)
             return false;
         }
 
-        if (strpos($route, '/') !== false) {
-            list($id, $route) = explode('/', $route, 2);
-        } else {
-            $id = $route;
-            $route = '';
+        // module and controller map take precedence
+        $routeInfo=pathinfo($route);
+        $id =$routeInfo['dirname']!=='.'?$routeInfo['dirname']:$routeInfo['basename'];
```

Comment:
Why is there a dot in a route?

Reply:
Because if `$route` is an empty string `""`  the `pathinfo($route)['dirname']` returns `"."`

Reply:
That certainly looks very magical. Can we solve it without `pathinfo()`?

Reply:
Absolutely yes.
```
 if (($pos = strrpos($route, '/')) !== false) {
            $id = substr($route, 0, $pos);
            $route = substr($route, $pos + 1);
        }else{
            $id = $route;
            $route = '';
}
```
give completely same result as: 

```
$routeInfo=pathinfo($route);
$id =$routeInfo['dirname']!=='.'?$routeInfo['dirname']:$routeInfo['basename'];
$route = $routeInfo['dirname']!=='.'?$routeInfo['basename']:'';
```
but I thought calling only one function `pathinfo()`  might perform better than calling `strrpos()` and `substr()` twice!!?